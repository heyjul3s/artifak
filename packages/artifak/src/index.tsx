export {
  createStyledComponent,
  createBaseComponents
} from '@artifak/component-generator';

export { BlockBase, createBlockComponents } from '@artifak/block';
export { BlockBaseProps } from '@artifak/block';

export { FlexRow, FlexRowBase, FlexCol, FlexColBase } from '@artifak/flex';
export { FlexRowBaseProps, FlexColBaseProps } from '@artifak/flex';

export { Grid, GridBase, GridItem, GridItemBase } from '@artifak/grid';
export { GridBaseProps, GridItemBaseProps } from '@artifak/grid';

export {
  createTypographyComponents,
  fluidSizing,
  fontWeight,
  TypographyBase
} from '@artifak/typography';
export { TypographyBaseProps } from '@artifak/typography';

export { Imagery, ImageryBase, ImageryProps } from '@artifak/imagery';

export { media, Media } from '@artifak/media';
