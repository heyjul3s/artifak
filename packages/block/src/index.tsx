export { createBlockComponents } from './createBlockComponents';
export { BlockBase } from './BlockBase';
export { BlockBaseProps } from './typings';
