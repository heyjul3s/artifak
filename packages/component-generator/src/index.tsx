export { createComponents } from './createComponents';
export { createStyledComponent } from './createStyledComponent';
export {
  Attrs,
  BaseProps,
  BaseComponentProps,
  GenericRecord,
  Settings,
  StyledComponentConfig,
  Variant
} from './typings';
