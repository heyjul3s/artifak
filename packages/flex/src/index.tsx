export { FlexRow } from './FlexRow';
export { FlexRowBase } from './FlexRowBase';
export { FlexCol } from './FlexCol';
export { FlexColBase } from './FlexColBase';
export { FlexColBaseProps, FlexRowBaseProps } from './typings';
