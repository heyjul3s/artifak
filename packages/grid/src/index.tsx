export { Grid } from './Grid';
export { GridItem } from './GridItem';
export { GridBase } from './GridBase';
export { GridItemBase } from './GridItemBase';
export { GridBaseProps, GridItemBaseProps } from './typings';
