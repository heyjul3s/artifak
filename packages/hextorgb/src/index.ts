export { hexToRGB, getRGBvalues } from './hextorgb';
export { RGBColor } from './typings';
