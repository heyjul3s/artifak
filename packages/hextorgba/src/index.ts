export { hexToRGBA } from './hextorgba';
export { getHexColorRGBvalues } from '@artifak/hextorgb';
