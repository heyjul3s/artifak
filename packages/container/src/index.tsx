import React from 'react';
import { createStyledComponent, BaseProps } from '@artifak/component-generator';

export function Container({ children, ...props }) {
  return <ContainerBase {...props}>{children}</ContainerBase>;
}

const ContainerBase = createStyledComponent<BaseProps<any>>({
  styles: {
    margin: '0 auto',
    width: '100%'
  }
});
