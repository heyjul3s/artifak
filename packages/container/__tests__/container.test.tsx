import { Container } from '../src';
import React from 'react';
import { render } from '@testing-library/react';

describe('@artifak/container', () => {
  it('renders', () => {
    const { asFragment } = render(
      <Container>
        <>
          <p>Hello World</p>
        </>
      </Container>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders', () => {
    const { asFragment } = render(
      <Container maxWidth={['40em', '52em', '64em', '80em']}>
        <>
          <p>Hello World</p>
        </>
      </Container>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
