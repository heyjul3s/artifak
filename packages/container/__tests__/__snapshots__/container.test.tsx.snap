// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@artifak/container renders 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ eaQfTv"
    width="100%"
  >
    <p>
      Hello World
    </p>
  </div>
</DocumentFragment>
`;

exports[`@artifak/container renders 2`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ lmFeAU"
    width="100%"
  >
    <p>
      Hello World
    </p>
  </div>
</DocumentFragment>
`;
